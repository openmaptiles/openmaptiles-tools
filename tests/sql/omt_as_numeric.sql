SELECT omt_as_numeric(null) AS fail_null;
SELECT omt_as_numeric('.') AS fail_dot;
SELECT omt_as_numeric('a123') AS fail_prefixed_letter;
SELECT omt_as_numeric('123a') AS fail_postfixed_letter;
SELECT omt_as_numeric('foobar') AS fail_text;
SELECT omt_as_numeric('9999999999') AS ok_big_positive;
SELECT omt_as_numeric('-9999999999') AS ok_big_negative;
SELECT omt_as_numeric('123') AS ok_integer;
SELECT omt_as_numeric('123.456') AS ok_float;
SELECT omt_as_numeric('  123.456  ') AS ok_float_trim;
SELECT omt_as_numeric('456.789e2') AS ok_e_notation;
