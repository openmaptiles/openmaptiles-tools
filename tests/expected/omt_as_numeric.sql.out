 fail_null 
-----------
        -1
(1 row)

 fail_dot 
----------
       -1
(1 row)

 fail_prefixed_letter 
----------------------
                   -1
(1 row)

 fail_postfixed_letter 
-----------------------
                    -1
(1 row)

 fail_text 
-----------
        -1
(1 row)

 ok_big_positive 
-----------------
      9999999999
(1 row)

 ok_big_negative 
-----------------
     -9999999999
(1 row)

 ok_integer 
------------
        123
(1 row)

 ok_float 
----------
  123.456
(1 row)

 ok_float_trim 
---------------
       123.456
(1 row)

 ok_e_notation 
---------------
       45678.9
(1 row)

